timezone/src/lib/timezone/ntp_adapter.rb

```diff
@@ -0,0 +1,476 @@
+# encoding: utf-8
+module Yast
+  # An interface of the timezone and time setting package
+  # to the NTP package.
+  class NtpAdapterClient < Client
+    def main
+      Yast.import "UI"
+      # FIXME: only change this when new translations are done
+      # textdomain "country"
+      textdomain "ntp-client"
+
+
+      Yast.import "Address"
+      Yast.import "NetworkService"
+      Yast.import "NtpClient"
+      Yast.import "Service"
+      Yast.import "String"
+      Yast.import "Stage"
+      Yast.import "PackageSystem"
+      Yast.import "Popup"
+      Yast.import "Progress"
+      Yast.import "Report"
+      Yast.import "Timezone"
+      Yast.import "Wizard"
+
+      #     API:
+      #
+      # Usual *_proposal functions: MakeProposal, AskUser, Write.
+      # (but not Description; see, it just *looks* like *_proposal)
+      # Additionally:
+      #  GetNTPEnabled  (queries Service::Enabled)
+      #  SetUseNTP [ntp_used]
+      @ret = nil
+      @func = ""
+      @param = {}
+
+      if Ops.greater_than(Builtins.size(WFM.Args), 0) &&
+          Ops.is_string?(WFM.Args(0))
+        @func = Convert.to_string(WFM.Args(0))
+        if Ops.greater_than(Builtins.size(WFM.Args), 1) &&
+            Ops.is_map?(WFM.Args(1))
+          @param = Convert.to_map(WFM.Args(1))
+        end
+      end
+
+      Builtins.y2milestone(
+        "ntp-client_proposal called func %1 param %2",
+        @func,
+        @param
+      )
+
+      # FIXME must go to module to preserve value
+      @ntp_was_used = false
+
+
+
+      if false
```

Comment:
NP: This `if false` is useless, and the rest could be replaced by `case...when` as well...